clickhouse/base/socket.cpp

```diff
@@ -151,6 +151,19 @@ SocketHolder::operator SOCKET () const noexcept {
     return handle_;
 }
 
+void SocketHolder::SetTcpKeepAlive(int idle, int intvl, int cnt) noexcept {
+    int val = 1;
+    setsockopt(handle_, SOL_SOCKET, SO_KEEPALIVE, &val, sizeof val);
+
+#if defined(_linux_)
```

Comment:
I think the same options may be applied in the case of macOS too, so I suggest to use \_unix_ instead of \_linux_.